neon-tracer/src/data_source/neon_api.rs

```
@@ -116,46 +110,35 @@ impl NeonAPIDataSource {
             .get_storage_at(get_storage_at_request, id)
             .await
             .map(|GetStorageAtReturn(arr)| U256::from_be_bytes(arr))
-            .map_err(|e| Error::owned(ErrorCode::InternalError.code(), e.to_string(), None::<()>))
+            // `context` cannot be used here because `NeonAPIClientError` isn't Send/Sync.
+            .map_err(|e| anyhow!("neon_api::get_storage_at failed: {:?}", e))
     }
 
     pub async fn get_balance(&self, get_ether_request: GetEtherRequest, id: u64) -> Result<U256> {
-        let response = self
+        match self
             .api_client
             .get_ether_account_data(get_ether_request, id)
-            .await;
-
-        if response.is_err() {
-            Ok(U256::default())
-        } else {
-            U256::from_str_prefixed(&response.unwrap().balance)
-                .map_err(|_| ERR("cast GetEtherAccountDataReturn.balance->U256", id))
+            .await
+        {
+            Err(_) => Ok(U256::default()),
+            Ok(r) => U256::from_str_prefixed(&r.balance)
+                .context("Invalid balance returned by neon_api::get_ether_account_data"),
         }
     }
 
-    pub async fn get_trx_count(&self, get_ether_request: GetEtherRequest, id: u64) -> Result<U256> {
-        let response = self
-            .api_client
+    pub async fn get_trx_count(&self, get_ether_request: GetEtherRequest, id: u64) -> U256 {
+        self.api_client
             .get_ether_account_data(get_ether_request, id)
-            .await;
-
-        if response.is_err() {
-            Ok(U256::default())
-        } else {
-            Ok(U256::new(response.unwrap().trx_count.into()))
-        }
+            .await
+            .map(|r| U256::new(r.trx_count.into()))
+            .unwrap_or_default()
```

Let's bubble up the error instead of shadowing it into a default value.